scipy/stats/tests/test_distributions.py

```diff
@@ -913,6 +913,93 @@ def test_sf(self):
+                                loc=(rvs_loc + 2))
+        shape_mle, loc_mle, scale_mle = stats.pareto.fit(data, floc=2)
+        assert_equal(scale_mle + 2, data.min())
+        assert_equal(shape_mle, 1/((1/len(data - 2)) *
+                                   np.sum(np.log((data
+                                                  - 2)/(data.min() - 2)))))
+        assert_equal(loc_mle, 2)
+
+    @pytest.mark.filterwarnings("ignore:invalid value encountered in "
+                                "double_scalars")
+    @pytest.mark.parametrize("rvs_shape", [1, 2])
+    @pytest.mark.parametrize("rvs_loc", [0, 2])
+    @pytest.mark.parametrize("rvs_scale", [1, 5])
+    def test_fit_MLE_comp_optimzer(self, rvs_shape, rvs_loc, rvs_scale):
+        data = stats.pareto.rvs(size=100, b=rvs_shape, scale=rvs_scale,
+                                loc=rvs_loc)
+        args = [data, (stats.pareto._fitstart(data), )]
+        func = stats.pareto._reduce_func(args, {})[1]
+
+        # fixed `floc` to actual location provides as good or better fit.
+        shape_mle, loc_mle, scale_mle = stats.pareto.fit(data, floc=rvs_loc)
+        shape_opt, loc_opt, scale_opt = super(type(stats.pareto),
+                                              stats.pareto).fit(data,
+                                                                floc=rvs_loc)
+        ll_mle = func((shape_mle, loc_mle, scale_mle), data)
+        ll_opt = func((shape_opt, loc_opt, scale_opt), data)
+        assert ll_mle < ll_opt or np.allclose(ll_mle, ll_opt)
+
+        # fixing `floc` to an arbitrary number, 0, still provides an as good
+        # or better fit.
+        shape_mle, loc_mle, scale_mle = stats.pareto.fit(data, floc=0)
+        shape_opt, loc_opt, scale_opt = super(type(stats.pareto),
+                                              stats.pareto).fit(data, floc=0)
+        ll_mle = func((shape_mle, loc_mle, scale_mle), data)
+        ll_opt = func((shape_opt, loc_opt, scale_opt), data)
+        assert ll_mle < ll_opt or np.allclose(ll_mle, ll_opt)
+
+        # fixed shape still uses MLE and provides an as good or better fit.
+        shape_mle, loc_mle, scale_mle = stats.pareto.fit(data, floc=0, f0=4)
+        shape_opt, loc_opt, scale_opt = super(type(stats.pareto),
+                                              stats.pareto).fit(data,
+                                                                floc=0, f0=4)
+        ll_mle = func((shape_mle, loc_mle, scale_mle), data)
+        ll_opt = func((shape_opt, loc_opt, scale_opt), data)
+        assert ll_mle < ll_opt or np.allclose(ll_mle, ll_opt)
+
+    def test_fit_warnings(self):
+        assert_raises(RuntimeError, stats.pareto.fit, [1, 2, 3], f0=2, floc=1,
+                      fscale=1)
+        assert_raises(RuntimeError, stats.pareto.fit, [np.nan])
+        assert_raises(RuntimeError, stats.pareto.fit, [np.inf])
+        assert_raises(TypeError, stats.pareto.fit, [2, 2, 3], floc=1, extra=2)
```

Comment:
We'll need to check for the error when the user's `floc` makes the data go out of range.

Reply:
Agree - it has proved to be difficult to access the `FitDataError` outside of `_continuous_distn`.